## Question 1
### Rocket_Motor

In [1]:
import pandas as pd
import numpy as np
from sklearn.linear_model import LinearRegression
import statsmodels.api as sm

#### Fitting of Simple Linear Regression Model

In [2]:
# Load data into a pandas DataFrame
data =pd.read_csv('Rocket_motor_data.csv', index_col=0)
data

Shear Strength y (psi)  Age of Propellant x (weeks)
Observation i                                                     
1                             2158.70                        15.50
2                             1678.15                        23.75
3                             2316.00                         8.00
4                             2061.30                        17.00
5                             2207.50                         5.50
6                             1708.30                        19.00
7                             1784.70                        24.00
8                             2575.00                         2.50
9                             2357.90                         7.50
10                            2236.70                        11.00
11                            2165.20                        13.00
12                            2399.55                         3.75
13                            1779.80                        25.00
14                            2336.75                         9.75
15                            1765.30                        22.00
16                            2053.50                        18.00
17                            2414.40                         6.00
18                            2200.50                        12.50
19                            2654.20                         2.00
20                            1753.70                        21.50

In [3]:
X = data['Age of Propellant x (weeks)'].values.reshape(-1, 1)
y = data['Shear Strength y (psi)'].values

# Fit linear regression model
model = LinearRegression()
model.fit(X, y)
print("Intercept= ", round(model.intercept_, 4))
print("Slope= ", round(model.coef_[0], 4))

Intercept=  2626.2518
Slope=  -37.1109


#### Test of Hypothesis

In [20]:
# Add a constant to the predictor variable
X = sm.add_constant(X)

# Fit the linear regression model
model = sm.OLS(y, X).fit()

# Print the summary
print(model.summary())

                            OLS Regression Results                            
Dep. Variable:                      y   R-squared:                       0.902
Model:                            OLS   Adj. R-squared:                  0.897
Method:                 Least Squares   F-statistic:                     166.1
Date:                Sun, 24 Mar 2024   Prob (F-statistic):           1.58e-10
Time:                        11:17:41   Log-Likelihood:                -118.57
No. Observations:                  20   AIC:                             241.1
Df Residuals:                      18   BIC:                             243.1
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const       2626.2518     44.034     59.641      0.0

# Conclusion
## Shear Strength = 2626.25 - 37.11(Age of Propellant)

- Here the model tells us that 1 week change in Age of Propellant there will be an average decrease of 37.11 in Shear Strength, but when the Age of Propellant is zero or constant there will be an average increase of 2626.25 in the Shear Strength.
- The p-value which is less than 0.05 shows that there is a statistically significant associtation between the Shear Strength and Age of Propellant.
- R-squared value tells us that the 90.2% vairation can be explained by the Age of Propellant variable.
- Here, F statistics and corresponding p-values tell us the overall significance of the regression model that is whether explanatory variable in the model are useful for explaining the variation in response variable. Since p-value in the example is less than the 0.05 therefore our model is statistically significant, thus the regressor i.e, Age of Propellant is usefull in explaining the variation of the Shear Strength.


## Question 2
### Soft Drinks

In [4]:
import pandas as pd
import statsmodels.api as sm

In [6]:
df = pd.read_csv("soft_drink.csv", index_col='obs')
df

Delivery Time  Number of Cases  Distance
obs                                          
1            16.68                7       560
2            11.50                3       220
3            12.03                3       340
4            14.88                4        80
5            13.75                6       150
6            18.11                7       330
7             8.00                2       110
8            17.83                7       210
9            79.24               30      1460
10           21.50                5       605
11           40.33               16       688
12           21.00               10       215
13           13.50                4       255
14           19.75                6       462
15           24.00                9       448
16           29.00               10       776
17           15.35                6       200
18            9.50                3       132
19           19.00               17        36
20           35.10               17       770
21           17.90               10       140
22           52.32               26       810
23           18.75                9       450
24           19.83                8       635
25           10.75                4       150

In [ ]:
# Fit linear regression model
model = LinearRegression()
model.fit(X, y)
print("Intercept= ", round(model.intercept_, 4))
print("Slope= ", round(model.coef_[0], 4))

In [8]:
# Define the dependent variable and the independent variables
y = df['Delivery Time']
X = df[['Number of Cases', 'Distance']]

# Add a constant to the independent variables matrix
X = sm.add_constant(X)

# Fit the ordinary least squares (OLS) model
model = sm.OLS(y, X)
results = model.fit()

# Print the summary statistics of the regression model
print(results.summary())

                            OLS Regression Results                            
Dep. Variable:          Delivery Time   R-squared:                       0.952
Model:                            OLS   Adj. R-squared:                  0.948
Method:                 Least Squares   F-statistic:                     219.5
Date:                Sun, 24 Mar 2024   Prob (F-statistic):           2.93e-15
Time:                        15:10:02   Log-Likelihood:                -65.496
No. Observations:                  25   AIC:                             137.0
Df Residuals:                      22   BIC:                             140.6
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
                      coef    std err          t      P>|t|      [0.025      0.975]
-----------------------------------------------------------------------------------
const               1.5509      1.221     

In [9]:
# Print the confidence intervals for the model coefficients
print(results.conf_int(alpha=0.05))


                        0         1
const           -0.981496  4.083376
Number of Cases  1.014923  1.634601
Distance         0.014523  0.027982


##  = 1.55 + 1.32(Number of Cases) + 0.02(Distance)

- Here the model tells us that 1 week change in Age of Propellant there will be an average decrease of 37.11 in Shear Strength, but when the Age of Propellant is zero or constant there will be an average increase of 2626.25 in the Shear Strength.
- The p-value which is less than 0.05 shows that there is a statistically significant associtation between the Shear Strength and Age of Propellant.
- R-squared value tells us that the 90.2% vairation can be explained by the Age of Propellant variable.
- Here, F statistics and corresponding p-values tell us the overall significance of the regression model that is whether explanatory variable in the model are useful for explaining the variation in response variable. Since p-value in the example is less than the 0.05 therefore our model is statistically significant, thus the regressor i.e, Age of Propellant is usefull in explaining the variation of the Shear Strength.
